Model/Sync/Catalog/Processor.php

```diff
@@ -13,6 +13,7 @@
 use Magento\Framework\Stdlib\DateTime\DateTime;
 use Yotpo\Core\Model\Sync\Category\Processor\ProcessByProduct as CategorySyncProcessor;
 use Magento\Quote\Model\Quote;
+use Yotpo\Core\Model\Sync\Data\Main as SyncDataMain;
```

Comment:
We don't need to import Sync\Data\Main separately.
Yotpo\Core\Model\Sync\Catalog\Data is already a child class of  Sync\Data\Main. So any public functions from Sync\Data\Main can be called via CatalogData imported

Reply:
I understand that we can use `catalogData->getProductIds` like so:
```
return $this->catalogData->getProductIds($productIds, $storeId, $itemsMap);
```

But looking at these two imports, I don't understand why.
```
use Yotpo\Core\Model\Sync\Data\Main as SyncDataMain;
use Yotpo\Core\Model\Sync\Catalog\Data as CatalogData;
```

I see that `Yotpo\Core\Model\Sync\Data\Main` uses `Yotpo\Core\Model\Sync\Catalog\Data `.
But `Yotpo\Core\Model\Sync\Data\Main` also uses `Yotpo\Core\Model\Config`.

Why do `Model/Sync/Catalog/Processor.php` uses both `Yotpo\Core\Model\Sync\Data\Main` and `Yotpo\Core\Model\Config` then?

Reply:
Hi Eran,

I think your original question is the following.

Why do Model/Sync/Catalog/Processor.php uses both Yotpo\Core\Model\Sync\Catalog\Data and Yotpo\Core\Model\Config then? 

Please find my response below.

The arguments specified in a class constructor are the dependencies for that class who owns that constructor. The access scope of these dependencies are limited to that owner class only.

For example, Model\Sync\Catalog\Processor wants to use some methods defined at Yotpo\Core\Model\Config class . So we can say Model\Sync\Catalog\Processor has a dependency on Yotpo\Core\Model\Config . So we should inject it as a parameter in the constructor of Model\Sync\Catalog\Processor class .

In the same way,

Yotpo\Core\Model\Sync\Catalog\Data wants to use some methods defined at Yotpo\Core\Model\Config . So Model\Sync\Catalog\Data has a dependency on Yotpo\Core\Model\Config . So we should inject it in the constructor of Model\Sync\Catalog\Data class .

Here Yotpo\Core\Model\Sync\Catalog\Data and Model\Sync\Catalog\Processor can not share Config class between them.

Actually Yotpo\Core\Model\Sync\Catalog\Data is another dependency of Model\Sync\Catalog\Processor class (like the config class). That is why we injected Yotpo\Core\Model\Sync\Catalog\Data inside the constructor of Processor class.

The reason why I suggested to avoid the separate inclusion of Yotpo\Core\Model\Sync\Data\Main at Processor class is because of the following.

Processor class is already containing Yotpo\Core\Model\Sync\Catalog\Data as it's dependecy. Yotpo\Core\Model\Sync\Catalog\Data is actually a child class of Yotpo\Core\Model\Sync\Data\Main. So any methods or properties defined at Yotpo\Core\Model\Sync\Data\Main will be available at Yotpo\Core\Model\Sync\Catalog\Data also. We can then call all those properties from  Yotpo\Core\Model\Sync\Catalog\Data itself.

Hope you are clear on this. Sorry if my response added more confusions. Please let me know your thoughts.

Reply:
Thanks for the details @manaftryzens. It is clear now.